sfm_pc/management/commands/import_locations.py

```diff
@@ -0,0 +1,142 @@
+import urllib.request
+import subprocess
+import os
+
+from django.core.management.base import BaseCommand, CommandError
+from django.db import connection
+from django.conf import settings
+
+
+class Command(BaseCommand):
+    help = 'Import raw location data into Location objects'
+
+    TABLE_NAME = 'canonical_locations'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--location_file',
+            dest='location_file',
+            default='fixtures/locations.geojson',
+            help='Relative path to location file for import'
+        )
+
+    def handle(self, *args, **options):
+
+        self.location_file = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)),
+            '..',
+            '..',
+            '..',
+            options['location_file'],
+        )
+
+        self.import_raw_locations()
+        self.create_location_objects()
+
+    def import_raw_locations(self):
+        with connection.cursor() as cursor:
+            cursor.execute('DROP TABLE IF EXISTS {}'.format(self.TABLE_NAME))
+            self.stdout.write('Dropped existing table "{}"'.format(self.TABLE_NAME))
+
+        process = subprocess.Popen([
+            'ogr2ogr',
+            '-f',
+            'PostgreSQL',
+            'PG:dbname={NAME} user={USER} password={PASSWORD} host={HOST} port={PORT}'.format(**settings.DATABASES['default']),
+            self.location_file,
+            '-nln',
+            self.TABLE_NAME,
+        ], stdout=subprocess.PIPE)
+
+        output, error = process.communicate()
+
+        if output:
+            self.stdout.write(self.style.SUCCESS(output.decode('utf-8')))
+
+        if error:
+            self.stdout.write(self.style.ERROR(error.decode('utf-8')))
+            raise CommandError('Could not import raw location file')
+
+        with connection.cursor() as cursor:
+            for column in ('sfm', 'tags'):
+                cursor.execute('''
+                    ALTER TABLE {table_name}
+                    ALTER COLUMN {column_name}
+                    TYPE json USING {column_name}::json
+                '''.format(table_name=self.TABLE_NAME,
+                           column_name=column))
+
+                cursor.execute('SELECT COUNT(*) FROM {}'.format(self.TABLE_NAME))
+
+                row = cursor.fetchone()
+
+                n_inserted, = row
+
+        if not error:
+            self.stdout.write(
+                self.style.SUCCESS(
+                    'Imported {} records from raw location file'.format(n_inserted)
+                )
+            )
+
+    def create_location_objects(self):
+        insert = '''
+            INSERT INTO location_location (
+              id,
+              name,
+              feature_type,
+              division_id,
+              tags,
+              sfm,
+              adminlevel,
+              adminlevel1_id,
+              adminlevel2_id,
+              geometry
+            )
+            SELECT
```

Comment:
i'd consider a CTE here to make this a little more self-documenting, but do not require that change

Reply:
Cool, good thought. I gave the joined tables more helpful aliases.